File: test-samples/python/lite-LLM.py
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Import required libraries AFTER loading .env
import litellm


# Ensure REVENIUM_METERING_API_KEY and the relevant provider API key
# (e.g., OPENAI_API_KEY) are set in your .env file

response = litellm.completion(
    model="gpt-4o-mini", # Ensure this model's provider key is in .env
    messages=[{ "content": "Please verify you are ready to assist me.","role": "user"}],

    # Optional metadata to send to Revenium for
    # advanced reporting. Uncomment lines
    # as needed for your use case.

    # usage_metadata={
        # "trace_id": "conv-28a7e9d4",
        # "task_type": "analyze-spectral-data",
        # "subscriber": {
        #   "id": "1473847563",
        #   "email": "carol@finoptic.com",
        #   "credential": {
        #     "name": "Engineering API Key",
        #     "value": "hak-abc123456"
        #   }
        # },
        # "organization_id": "Finoptic Labs",
        # "subscription_id": "sub_gold_1234567890",
        # "product_id": "spectral-analyzer-gold",
        # "agent": "chemistry-agent",
    # },

)
print(response.choices[0].message.content)

File: test-samples/node/lite-LLM.js
import 'dotenv/config';


async function main() {
  const response = await fetch(`${process.env.LITELLM_PROXY_URL}/chat/completions`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${process.env.LITELLM_API_KEY}`,
      // Optional metadata for enhanced tracking
      'x-revenium-subscriber-id': 'demo-user-123',
      'x-revenium-subscriber-email': 'demo-user@acme.com',
      'x-revenium-subscriber-credential-name': 'api-key',
      'x-revenium-subscriber-credential': 'demo-credential-value',
      'x-revenium-organization-id': 'revenium-demo',
      'x-revenium-task-type': 'chat-completion',
      'x-revenium-product-id': 'my-ai-app',
      'x-revenium-agent': 'openai-assistant'
    },
    body: JSON.stringify({
      model: 'gpt-4o-mini',
      messages: [
        {
          role: 'system',
          content: 'You are a helpful assistant.'
        },
        {
          role: 'user',
          content: 'Please verify you are ready to assist me.'
        }
      ]
    })
  });

  const data = await response.json();
  console.log(data.choices[0].message.content);
}

main().catch(console.error);